documentation/modules/proc-configuring-kafka-bridge-tracing.adoc

```diff
@@ -0,0 +1,103 @@
+// Module included in the following assemblies:
+//
+// assembly-kafka-bridge-config.adoc
+
+[id='proc-configuring-kafka-bridge-tracing-{context}']
+= Configuring distributed tracing
+
+[role="_abstract"]
+Enable distributed tracing to trace messages consumed and produced by the Kafka Bridge, and HTTP requests from client applications.
+
+Properties to enable tracing are present in the `application.properties` file.
+To enable distributed tracing, do the following:
+
+* Uncomment the required tracing properties in the properties file.
+* Set environment variables for tracing.
+
+With the default configuration, you can introduce a Jaeger tracing system based on OpenTelemetry or OpenTracing.
+
+OpenTelemetry and OpenTracing are API specifications for collecting tracing data as _spans_ of metrics data.
+Spans represent a specific operation.
+A trace is a collection of one or more spans.
+
+Traces are generated when:
+
+* The Kafka Bridge sends messages to HTTP clients and consumes messages from HTTP clients
+* HTTP clients send HTTP requests to send and receive messages through the Kafka Bridge
```

Comment:
aren't these two sentence saying the same but from two different point of view? bridge and clients? or did you mean something different?

Reply:
This is still what I see ...

`HTTP clients send HTTP requests to the Kafka Bridge` IS EQUAL TO `The Kafka Bridge consumes messages from HTTP clients`
`HTTP clients receive messages from the Kafka Bridge` IS EQUAL TO `The Kafka Bridge sends messages to HTTP clients`

But I think what you mean. I guess you are referring to the fact that if tracing is enabled on the HTTP client, and for example you are sending to the bridge, you will get: a span from the HTTP client (about its send), a span in the bridge about sending to Kafka, then finally a span from the Kafka instrumented library about message sent to Kafka. The same could apply for consuming in the other way around.
Anyway the spans coming from an HTTP client are not related to the tracing enabled on the bridge, but the tracing enabled on the client itself.
Or maybe you want to say something different which I am not getting right :-D

Reply:
Let's just make it simpler. :-)

> Traces are generated when the Kafka Bridge does the following:
> 
> * Sends messages to HTTP clients
> * Consumes messages from HTTP clients